src/main/java/ch/wisv/areafiftylan/products/controller/TicketRestController.java

```diff
@@ -119,13 +119,32 @@ public TicketRestController(TicketService ticketService, OrderService orderServi
     @PreAuthorize("hasRole('ADMIN')")
     @PostMapping("/types")
     public ResponseEntity<?> addTicketType(@RequestBody @Validated TicketType type) {
-
         TicketType ticketType = ticketService.addTicketType(type);
-
         return createResponseEntity(HttpStatus.CREATED, "TicketType successfully added", ticketType);
     }
 
     @PreAuthorize("hasRole('ADMIN')")
+    @GetMapping("/types")
+    public ResponseEntity<?> readTicketTypes() {
+        Collection<TicketType> ticketTypes = ticketService.getAllTicketTypes();
+        return createResponseEntity(HttpStatus.OK, "TicketTypes successfully retrieved.", ticketTypes);
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PutMapping("/types")
+    public ResponseEntity<?> updateTicketType(@RequestBody @Validated TicketType type) {
+        TicketType ticketType = ticketService.updateTicketType(type);
+        return createResponseEntity(HttpStatus.OK, "TicketType successfully updated.", ticketType);
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @DeleteMapping("/types")
```

Comment:
This implies deleting all TicketTypes. It should be a DELETE on `/types/{id}` or something like that.